pkgs/applications/science/math/mxnet/default.nix

```diff
@@ -1,9 +1,13 @@
 { config, stdenv, lib, fetchurl, fetchpatch, bash, cmake
 , opencv3, gtest, blas, gomp, llvmPackages, perl
-, cudaSupport ? config.cudaSupport or false, cudatoolkit, nvidia_x11
-, cudnnSupport ? cudaSupport, cudnn
+, cudaSupport ? config.cudaSupport or false, cudaPackages ? {}, nvidia_x11
```

Comment:
The function needs to be called with a `cudaPackages` parameter otherwise, even when `cudaSupport` is false. This is similar to the `foo ? null` that you can see in expressions.

Reply:
Yes, but `cudaPackages` would be part of `all-packages.nix`, wouldn't it?

Reply:
So, wouldn't `cudaPackages` from top-level be automatically passed with `callPackage`?
Another reason I'm asking this, is that we we don't do any error handling when inheriting attributes from `cudaPackages` down below. I'd rather just not allow not passing `cudaPackages`

Reply:
Yes, here, in the package set where you have that attribute. But the attribute doesn't always have to be there. Granted, it won't happen quickly here. The expression is simply incorrect if it always requires `cudaPackages`, *even* when you're not building with cuda support. You are correct the handling further down is not good. That I leave up to the maintainers of the expressions.

Reply:
> even when you're not building with cuda support

But this is how people have been handling `cudatoolkit` and apple sdk `frameworks` all this time. The attribute (`darwin.apple_sdk.frameworks` or `cudatoolkit`) is there, but it's never evaluated unless `cudaSupport` is `true`.

I'm not saying it's good, but I can see some... consistency